internal/resource/url.go

```diff
@@ -86,8 +87,12 @@ func Fetch(l *log.Logger, c *HttpClient, ctx context.Context, u url.URL) ([]byte
 
 // FetchWithHeader fetches a resource given a URL. If the resource is
 // of the http or https scheme, the provided header will be used when
-// fetching. The supported schemes are http, data, and oem.
+// fetching. The supported schemes are http, data, tftp, and oem.
 func FetchWithHeader(l *log.Logger, c *HttpClient, ctx context.Context, u url.URL, h http.Header) ([]byte, error) {
+	if u.Scheme == "tftp" {
```

Comment:
Do note that the `FetchWithHeader` function is already kinda hacky by that metric, it takes `data://` and `oem://` urls too.

> FetchWithHeader fetches a resource given a URL. If the resource is of the http or https scheme, the provided header will be used when fetching.

Think I should add a rename/refactor to this PR to address this?